includes/class-wp-document-revisions-validate-structure.php

```diff
@@ -0,0 +1,628 @@
+		}
+
+		if ( 6 === $params['code'] ) {
+			// Attachment file name not encoded.
+			$title     = get_post_field( 'post_title', $id );
+			$attach_id = $parm;
+			$attach    = get_post( $attach_id );
+
+			// get file name.
+			$file = get_attached_file( $attach_id );
+
+			$file = $this->check_document_folder( $file );
+
+			// get filename part.
+			$filename = pathinfo( $file, PATHINFO_FILENAME );
+
+			// revalidate input (late, but before any damage is done).
+			if ( preg_match( '/^[a-f0-9]{32}$/', $filename ) ) {
+				return new WP_Error( 'inconsistent_parms', __( 'Inconsistent data sent to Interface', 'wp-document-revisions' ) );
+			}
+
+			$new_name = md5( $title . microtime() );
+			$new_file = str_replace( $filename, $new_name, $file );
+			// phpcs:ignore WordPress.PHP.NoSilencedErrors.Discouraged
+			if ( @copy( $file, $new_file ) ) {
+				$name = get_post_meta( $attach_id, '_wp_attached_file', true );
+				update_post_meta( $attach_id, '_wp_attached_file', str_replace( $filename, $new_name, $name ), $name );
+				unlink( $file );
+			}
+		}
+
+		if ( 7 === $params['code'] ) {
+			// Attachment file in wrong location (media not document).
+			$title     = get_post_field( 'post_title', $id );
+			$attach    = $parm;
+			$attach_id = get_post( $attach );
+
+			// get file name.
+			$orig = get_attached_file( $attach );
+
+			// manipulate file as in serve_file process.
+			$file = $this->check_document_folder( $orig );
+
+			// revalidate input (late, but before any damage is done).
+			if ( $orig === $file || ! file_exists( $orig ) || file_exists( $file ) ) {
+				return new WP_Error( 'inconsistent_parms', __( 'Inconsistent data sent to Interface', 'wp-document-revisions' ) );
+			}
+			$file_dir = dirname( $file );
+			// Use copy and unlink because rename breaks streams.
+			// phpcs:disable WordPress.PHP.NoSilencedErrors.Discouraged
+			// Ensure directory exists.
+			if ( ! is_dir( $file_dir ) ) {
+				wp_mkdir_p( $file_dir );
+			}
+			if ( @copy( $orig, $file ) ) {
+				chmod( $file, 0664 );
+				unlink( $orig );
+				// get attachment metadata.
+				$meta = get_post_meta( $attach, '_wp_attachment_metadata', true );
+				if ( ! is_array( $meta ) || ! isset( $meta['sizes'] ) ) {
+					null;
+				} else {
+					// image name contains only file name and extension.
+					$orig_dir = trailingslashit( dirname( $orig ) );
+					$file_dir = trailingslashit( $file_dir );
+					// move files.
+					foreach ( $meta['sizes'] as $size => $sizeinfo ) {
+						if ( file_exists( $orig_dir . $sizeinfo['file'] ) ) {
+							// Use copy and unlink because rename breaks streams.
+							if ( @copy( $orig_dir . $sizeinfo['file'], $file_dir . $sizeinfo['file'] ) ) {
+								@chmod( $file_dir . $sizeinfo['file'], 0664 );
+								unlink( $orig_dir . $sizeinfo['file'] );
+							}
+						}
+					}
+				}
+			}
+			// phpcs:enable WordPress.PHP.NoSilencedErrors.Discouraged
+		}
+
+		if ( 8 === $params['code'] ) {
```

Comment:
It may be good to document the codes and their description someplace to have them all in one place outside the `if` statement's comments.

Reply:
OK